mobile-app/src/firebase/api.ts

```diff
@@ -255,20 +298,51 @@ export async function fetchUserInfo({
   userId,
 }: {
   userId: string;
-}): Promise<userT> {
-  return mockUsers[userId];
+}): Promise<userWithIdT> {
+  const userDocRef = doc(firestore, "users", userId);
+  const userDocSnap = await getDoc(userDocRef);
+  // replace this with
+  if (!userDocSnap.exists()) {
+    throw new Error(`No user found with ID: ${userId}`);
+  }
+  const userData = userSnapToUserWithIdT(userDocSnap);
+  return userData;
 }
 
 export async function searchUsersInDb({
   searchText,
 }: {
   searchText: string;
 }): Promise<allUsersInfoT> {
-  return Object.fromEntries(
-    Object.entries(mockUsers).filter(
-      ([, userData]) =>
-        userData.displayName.toLowerCase().includes(searchText.toLowerCase()) ||
-        userData.username.toLowerCase().includes(searchText.toLowerCase()),
-    ),
+  // const
+  const usersCollection = collection(firestore, "users");
+
+  // displayname match
+  const displayNameQuery = query(
+    usersCollection,
+    where("displayName", ">=", searchText.toLowerCase()),
+    where("displayName", "<=", searchText.toLowerCase() + "\uf8ff"), // high Unicode character used to simulate "starting with" behaviour
+  );
+  const displayNameSnapshot = await getDocs(displayNameQuery);
+
+  // username match
+  const usernameQuery = query(
+    usersCollection,
+    where("username", ">=", searchText.toLowerCase()),
+    where("username", "<=", searchText.toLowerCase() + "\uf8ff"),
   );
+  const usernameSnapshot = await getDocs(usernameQuery);
```

Comment:
Why not just put both in 1 query

users where (displayName >= searchText AND displayName <= searchText) OR (username>= searchText AND username <= searchText)

Reply:
had to add a composite index in firestore, not sure what that does but it works